src/main/scala/cromwell/engine/WorkflowActor.scala

```diff
@@ -2,34 +2,76 @@ package cromwell.engine
 
 import akka.actor.Actor
 import akka.actor.SupervisorStrategy.Stop
+import cromwell.binding.{Workflow, WdlBinding, WdlValue}
 
 /**
  * This model assumes workflow actors are transient, created once per workflow instance.
  */
 object WorkflowActor {
 
-  // This is hand-waving over the actual engine-level representation of a workflow.
-  type WorkflowModel = String
-
-  case class Construct(model: WorkflowModel)
+  case class Construct(binding: WdlBinding, inputs: Map[String, WdlValue])
   case object Constructed
 
+  /**
+   * @param badInputs A `Map` of locally qualified input names to a corresponding diagnostic message.
+   */
+  case class ConstructionFailed(badInputs: Map[String, String])
+
   case object Start
   case object Started
 
   // Note there is no Stop message defined here, Stop is from Akka.
   case object Stopped
+
+  case object Done
+
+  case class InvalidOperation(message: String)
 }
 
-// Represents the root of a single workflow instance, not a manager of multiple
-// workflows.
+/** Represents the root of a single workflow instance, not a manager of multiple
+  * workflows.
+  */
 class WorkflowActor extends Actor {
+
   import WorkflowActor._
 
+  // This var would be unnecessary in a FSM that #becomes constructed.
+  private var _constructed = false
+
+  /**
+   * Check inputs for presence and type compatibility.
+   * @param workflow Workflow
+   * @param actualInputs Map of <b>local</b> input names to `WdlValue`s
+   * @return Map of locally qualified input names to a corresponding diagnostic message.
+   */
+  def checkUnsatisfiedInputs(workflow: Workflow, actualInputs: Map[String, WdlValue]): Map[String, String] =
```

Comment:
I'm not sure I understand why this algorithm calls `localizeInputs`.  I think you'll get key collisions because two tasks can have the same parameter name